core/variant/variant_construct.cpp

```diff
@@ -318,6 +318,25 @@ void VariantInternal::object_assign(Variant *v, const Object *o) {
 	}
 }
 
+void VariantInternal::object_assign_do_not_ref(Variant *v, const Object *o) {
```

Comment:
I'm not too happy with amount of code duplication, so if it's an issue, I'll take suggestions on how to reduce redundancy.

Reply:
An alternative could be to keep using `object_assign`, but decrement the reference in the GDScript VM after the fact. That's hacky though :P

Otherwise a boolean flag.

Reply:
Maybe `object_assign_do_not_ref` is the hack and decrementing in the gdscript VM is the more appropriate.

After all, calling the native function increments the refcount because it's expected that the caller will hold a local `Ref`, which will later be destructed and unref. This might be an issue of `MethodBind::ptrcall` not being safe with `Ref`.